### ISSUE TITLE
`inst' uses the type name

### ISSUE THREAD

**Issue by racket-bug-submit** [samth/test-bugs#178]
###### Originally submitted by Eli Barzilay <eli at barzilay dot org> on: Sun Jan 22 21:36:01 -0500 2012

A student tried to use (inst + Number), and got a huge error message that named the (beginning of) the full type as the offending error.  This is bad, since the source code of the error is just `+', so it would be way better to just say something like

``` racket
  Type Checker: Cannot instantiate non-polymorphic type of `+'

in that case.
```
##### Release:

```
HEAD
```
##### Environment:

```
Any
```
###### This bug was converted from Gnats bug [12507](http://bugs.racket-lang.org/old/12507).

[anon-submit; eli at barzilay dot org]

**Comment by racket-bug-submit** [comment]
###### On Wed, 01 Feb 2012 17:24:44 -0500, samth assigned this bug to stamourv

type printing

**Comment by stamourv** [comment]
###### On Thu, 02 Feb 2012 16:44:31 -0500, stamourv at racket-lang dot org wrote:

A commit by stamourv@racket-lang.org has resolved this report: b194916ad6
